### DISCUSSION TITLE
Install behind a firewall no internet

### DISCUSSION THREAD

**Discussion by MayanNetwork** [k3s-io/k3s#6142] [Q&A]
Please provide guidance on how to install behind a firewall, i.e. no internet access.  I can download the binary after it is approved by IT, and copy over.  But I may need to tweak-in different components.
Is there a way to download and build?  Building may be tooo much, so are there other components I can download that I can copy over ?

Thanks.

**Comment by brandond** [comment]
Have you checked out the documentation? https://rancher.com/docs/k3s/latest/en/installation/airgap/

**Reply by MayanNetwork** [reply]
Thanks, I will check that out.

**Reply by MayanNetwork** [reply]
Succes following those instructions but needed to resolve an ERROR output.   As practice, I ran it on my VM Centos Stream 8 machine, which is connected to the Internet, but my target RHEL 8.x will not be. 
I assume I will find a way to get around the following ERRORs, e.g. copy the packages.  So I would be fine with closing this "issue", and leaving it as notes for others.  The ERROR is below.  Noteworthy below in bold.

I am also curious to see if air-gap will work rootless.  The instructions say to run sudo 

[root@osboxes Downloads]# INSTALL_K3S_SKIP_DOWNLOAD=true ./install.sh 2>&1 | tee install_ag_log.txt
[INFO] Skipping k3s download and verify
[INFO] Skipping installation of SELinux RPM
**[ERROR]** Failed to apply container_runtime_exec_t to /usr/local/bin/k3s, please install:
dnf install -y container-selinux
dnf install -y https://rpm.rancher.io/k3s/stable/common/centos/8/noarch/**k3s-selinux-0.4-1.el8.noarch.rpm**
  
I  followed the above suggestion.  And the subesequent retry succeed in loading.
[osboxes@osboxes Downloads]$ kubectl get pods
**WARN**[0000] Unable to read /etc/rancher/k3s/k3s.yaml, **please start server with --write-kubeconfig-mode** to modify kube config permissions 
**error:** error loading config file "/etc/rancher/k3s/k3s.yaml": open /etc/rancher/k3s/k3s.yaml: permission denied

I simply did chmod 777 

$ kubectl cluster-info
Kubernetes control plane is running at https://127.0.0.1:6443
CoreDNS is running at https://127.0.0.1:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
Metrics-server is running at https://127.0.0.1:6443/api/v1/namespaces/kube-system/services/https:metrics-server:https/proxy

[osboxes@osboxes k3s-1.23.5-k3s1]$ sudo service k3s status
Redirecting to /bin/systemctl status k3s.service
● k3s.service - Lightweight Kubernetes
   Loaded: loaded (/etc/systemd/system/k3s.service; enabled; vendor preset: disabled)
   Active: active (running) since Sat 2022-09-24 14:11:51 EDT; 2h 47min ago
     Docs: https://k3s.io
  Process: 6824 ExecStartPre=/sbin/modprobe overlay (code=exited, status=0/SUCCESS)
  Process: 6822 ExecStartPre=/sbin/modprobe br_netfilter (code=exited, status=0/SUCCESS)
  Process: 6819 ExecStartPre=/bin/sh -xc ! /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service (code=exited, status=0/SUC>
 Main PID: 6829 (k3s-server)
    Tasks: 115
   Memory: 491.2M
   CGroup: /system.slice/k3s.service
           ├─6829 /usr/local/bin/k3s server
           ├─6845 containerd -c /var/lib/rancher/k3s/agent/etc/containerd/config.toml -a /run/k3s/containerd/containerd.sock --sta>
           ├─7586 /var/lib/rancher/k3s/data/05cfd5aec8ddf622207749ef3eda0e0efa12d8900105fdac78815a8cd6c73685/bin/containerd-shim-r>
           ├─7592 /var/lib/rancher/k3s/data/05cfd5aec8ddf622207749ef3eda0e0efa12d8900105fdac78815a8cd6c73685/bin/containerd-shim-r>
           ├─7735 /var/lib/rancher/k3s/data/05cfd5aec8ddf622207749ef3eda0e0efa12d8900105fdac78815a8cd6c73685/bin/containerd-shim-r>
           ├─9713 /var/lib/rancher/k3s/data/05cfd5aec8ddf622207749ef3eda0e0efa12d8900105fdac78815a8cd6c73685/bin/containerd-shim-r>
           └─9789 /var/lib/rancher/k3s/data/05cfd5aec8ddf622207749ef3eda0e0efa12d8900105fdac78815a8cd6c73685/bin/containerd-shim-r>

Sep 24 16:57:53 osboxes k3s[6829]: E0924 16:57:53.234091    6829 resource_quota_controller.go:413] **unable to retrieve** the complete>
Sep 24 16:57:53 osboxes k3s[6829]: W0924 16:57:53.740372    6829 handler_proxy.go:104] no RequestInfo found in the context
Sep 24 16:57:53 osboxes k3s[6829]: E0924 16:57:53.740487    6829 controller.go:116] loading OpenAPI spec for "v1beta1.metrics.k8s.>
Sep 24 16:57:53 osboxes k3s[6829]: , Header: map[Content-Type:[text/plain; charset=utf-8] X-Content-Type-Options:[nosniff]]
